### PR TITLE
Document clangd's type hierarchy extension

### PR DESCRIPTION
**Pull request by HighCommander4**
Fixes https://github.com/clangd/clangd/issues/808

### CODE REVIEW

**Comment by HighCommander4** [comment]
I didn't describe the protocol in detail (e.g. listing the new messages and their parameters), because it's described well in the linked LSP protocol and it seemed redundant to repeat that.

File: extensions.md [hunk 1]
```diff
@@ -167,6 +167,26 @@ clangd allows clients to use UTF-8 offsets instead. This allows clients that alw
...
+Type hierarchy allows the server to provide the client with information about
+the subclasses (derived classes) and superclasses (base classes) of the class
+referenced at a provided document location. Clients typically use this to show
+a tree view of the class and its subclasses or superclasses (or sometimes, in
+languages with single inheritance, a single tree showing both).
+
+The version of the protocol currently implemented is the one in
+[this proposal](https://github.com/microsoft/vscode-languageserver-node/pull/426).
+
+As of 2021, there is a
```
Comment (kadircet) [inline extensions.md:184]: I think we can drop this paragraph, but we should probably create a bug in clangd issue tracker to monitor the progress and migrate to the official protocol once it lands.

**Review by kadircet — APPROVED** [review]
thanks, lgtm!

File: extensions.md (hunk 1 above)
Comment (HighCommander4) [reply extensions.md]: Done, filed https://github.com/clangd/clangd/issues/826.

Could you merge the PR please (I don't have access)?

**Pull request merged**